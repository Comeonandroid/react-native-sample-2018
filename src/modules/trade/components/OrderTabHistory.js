import React, { PureComponent } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import { View, ListTable } from '../../../elements'

class OrderTabHistory extends PureComponent {

  componentDidMount() {
    this.props.getTradeOrders()
  }

  render() {
    const tableHead = ['Time', 'Pair', 'Type', 'Price (WETH)', 'Total (WETH)', 'Fee (ZRX)']
    return (
      <Container>
        <ListTable
          width={540}
          data={this.props.orders}
          header={tableHead}
          flexAr={[2, 2, 1, 2, 2, 2, 1]}
        />
      </Container>
    )
  }
}

OrderTabHistory.propTypes = {
  orders: PropTypes.array,
  getTradeOrders: PropTypes.func
}

OrderTabHistory.defaultProps = {
  orders: []
}

const Container = styled(View) ``

export default OrderTabHistory