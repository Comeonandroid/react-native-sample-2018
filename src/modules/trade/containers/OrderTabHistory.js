import { connect } from 'react-redux'

import OrderTabHistory from '../components/OrderTabHistory'

import { selectTradeOrders, getTradeOrders } from '../../../redux/trade'

const mapStateToProps = (state) => ({
  orders: selectTradeOrders(state.trade)
})

const mapDispatchToProps = {
  getTradeOrders
}

export default connect(mapStateToProps, mapDispatchToProps)(OrderTabHistory)
