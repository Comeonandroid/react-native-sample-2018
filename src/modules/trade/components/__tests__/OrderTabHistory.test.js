import 'react-native'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import { itBehavesLike } from '../../../../specs/shared_examples/index'

import OrderTabHistory from '../OrderTabHistory'

const initialState = { orders: [] }
const mockStore = configureStore()

const setup = propOverrides => {
  const store = mockStore(initialState)
  const props = Object.assign({
    getTradeOrders: jest.fn(),
    orders: []
  }, propOverrides)

  const component = <Provider store={store}><OrderTabHistory {...props} /></Provider>

  return {
    props,
    component
  }
}

describe('<OrderTabHistory />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })
})
