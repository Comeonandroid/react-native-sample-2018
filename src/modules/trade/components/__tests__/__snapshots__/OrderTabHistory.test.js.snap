// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderTabHistory /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {},
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "height": "100%",
          },
          undefined,
        ],
      ]
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          scrollWidth={undefined}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "position": "relative",
                  "width": "auto",
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {},
                undefined,
              ]
            }
          >
            <View
              data={
                Array [
                  "Time",
                  "Pair",
                  "Type",
                  "Price (WETH)",
                  "Total (WETH)",
                  "Fee (ZRX)",
                ]
              }
              flexAr={
                Array [
                  2,
                  2,
                  1,
                  2,
                  2,
                  2,
                  1,
                ]
              }
              header={true}
              style={
                Array [
                  Object {},
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#000000",
                      "display": "flex",
                      "flexDirection": "row",
                      "flexWrap": "nowrap",
                      "paddingBottom": 5,
                      "paddingLeft": 5,
                      "paddingRight": 5,
                      "paddingTop": 5,
                    },
                    undefined,
                  ],
                ]
              }
              width={540}
            >
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Time"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fcfcfc",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Time
                </Text>
              </View>
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Pair"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fcfcfc",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Pair
                </Text>
              </View>
              <View
                flexItem={1}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Type"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fcfcfc",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Type
                </Text>
              </View>
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Price (WETH)"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fcfcfc",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Price (WETH)
                </Text>
              </View>
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Total (WETH)"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fcfcfc",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Total (WETH)
                </Text>
              </View>
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Fee (ZRX)"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fcfcfc",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Fee (ZRX)
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {},
                  Array [
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "#282828",
                      "borderTopWidth": 1,
                      "height": 5,
                    },
                    undefined,
                  ],
                ]
              }
            />
          </View>
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            invertStickyHeaders={undefined}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`<OrderTabHistory /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {},
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "height": "100%",
          },
          undefined,
        ],
      ]
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          scrollWidth={undefined}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "position": "relative",
                  "width": "auto",
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {},
                undefined,
              ]
            }
          >
            <View
              data={
                Array [
                  "Time",
                  "Pair",
                  "Type",
                  "Price (WETH)",
                  "Total (WETH)",
                  "Fee (ZRX)",
                ]
              }
              flexAr={
                Array [
                  2,
                  2,
                  1,
                  2,
                  2,
                  2,
                  1,
                ]
              }
              header={true}
              style={
                Array [
                  Object {},
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fefaec",
                      "display": "flex",
                      "flexDirection": "row",
                      "flexWrap": "nowrap",
                      "paddingBottom": 5,
                      "paddingLeft": 5,
                      "paddingRight": 5,
                      "paddingTop": 5,
                    },
                    undefined,
                  ],
                ]
              }
              width={540}
            >
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Time"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Time
                </Text>
              </View>
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Pair"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Pair
                </Text>
              </View>
              <View
                flexItem={1}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Type"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Type
                </Text>
              </View>
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Price (WETH)"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Price (WETH)
                </Text>
              </View>
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Total (WETH)"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Total (WETH)
                </Text>
              </View>
              <View
                flexItem={2}
                style={
                  Array [
                    Object {},
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexBasis": 0,
                        "flexGrow": 2,
                        "flexShrink": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ],
                  ]
                }
                width={540}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  bold={true}
                  data="Fee (ZRX)"
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Open Sans",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "700",
                          "lineHeight": 15,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Fee (ZRX)
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {},
                  Array [
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "#5d5959",
                      "borderTopWidth": 1,
                      "height": 5,
                    },
                    undefined,
                  ],
                ]
              }
            />
          </View>
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            invertStickyHeaders={undefined}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
